Keas.Mvc/ClientApp/components/Equipment/AssignEquipment.tsx

```diff
@@ -1,87 +1,184 @@
-  modal: boolean;
   person: IPerson;
+  date: any;
+  error: string;
+  validState: boolean;
 }
 
 export default class AssignEquipment extends React.Component<IProps, IState> {
-  public static contextTypes = {
-    fetch: PropTypes.func,
-    person: PropTypes.object,
-    team: PropTypes.object
-  };
-  public context: AppContext;
+    public static contextTypes = {
+        fetch: PropTypes.func,
+        person: PropTypes.object,
+        team: PropTypes.object
+    };
+    public context: AppContext;
   constructor(props) {
     super(props);
     this.state = {
-      modal: false,
-      person: null
+      person: null,
+      date: moment().add(3, 'y'),
+      error: "",
+      validState: false,
     };
   }
 
-  public toggle = () => {
+  //clear everything out on close
+  private _closeModal = () => {
     this.setState({
-      modal: !this.state.modal
-    });
+        error: "",
+        validState: false,
+      });
+      this.props.closeModal();
   };
 
-  public createEquipment = async () => {
+  // assign the selected equipment even if we have to create it
+  private _assignSelected = async () => {
+
+    if (!this.state.validState) return;
+
     const person = this.context.person
-      ? this.context.person
-      : this.state.person;
-
-    const equipment = {
-      id: 0,
-      name: "newequipment" + new Date().getUTCSeconds(),
-      serialNumber: "SN123",
-      teamId: 1
-    };
+        ? this.context.person
+        : this.state.person;
+
+    await this.props.onCreate(person);
+
+    this._closeModal();
+  };
+
+  // once we have either selected or created the equipment we care about
+  private _onSelected = (equipment: IEquipment) => {
+      console.log("selected in assign", equipment);
+      //if this equipment is not already assigned
+
+      //TODO: more validation of name
+      if (equipment.name.length > 64)
+      {
+          this.props.selectEquipment(null);
+          this.setState({ error: "The equipment name you have chosen is too long" }, this._validateState);
+      }
+      //else if (this.props.assignedEquipmentList.findIndex(x => x == equipment.name) != -1)
+      //{
+      //    this.setState({ selectedEquipment: null, error: "The equipment you have chosen is already assigned to this user", validEquipment: false }, this._validateState);
+      //}
+      else
+      {
+          this.props.selectEquipment(equipment);
+          this.setState({ error: ""}, this._validateState);
+      }
+  };
 
-    await this.props.onCreate(equipment, person);
+  private _onDeselected = () => {
+      this.props.selectEquipment(null);
+      this.setState({ error: ""}, this._validateState);
+  }
 
-    // TODO: check for success
-    this.setState({ modal: false });
+  private _onSelectPerson = (person: IPerson) => {
+      this.setState({ person }, this._validateState);
   };
+
+
+  private _validateState = () => {
+      let valid = true;
+      if (this.props.selectedEquipment == null)
```

Comment:
@laholstege for these, try to use truthy values, like `if(!this.props.selectedEquipment)`.  That way it'll catch things like undefined.  Also, try to always use === or !== when comparing, so we don't get javascript trying to do some explicit conversions (like with the error message)